src/tools/rados/rados.cc

```diff
@@ -2750,8 +2752,30 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
     }
     else
       ret = 0;
-    string s(bl.c_str(), bl.length());
-    cout << s;
+
+    if (attr_name == "refcount")  {
+      obj_refcount oref;
+      auto p = bl.cbegin();
+      decode(oref, p);
+      for (auto itr = oref.refs.begin(); itr != oref.refs.end(); itr++) {
+	if (!itr->first.empty()) {
+	  cout << itr->first << "::" << itr->second << std::endl;
+	}
+	else {
+	  cout << "wildcard reference::" << itr->second << std::endl;
+	}
+      }
+      if (!oref.retired_refs.empty()) {
+	cout << "--------------------------------------" << std::endl;
+	for (const auto & ref : oref.retired_refs) {
+	  cout << "retired_refs::" << ref << std::endl;
+	}
+      }
+    }
+    else {
+      string s(bl.c_str(), bl.length());
+      cout << s << std::endl;
```

Comment:
The last `std::endl` is superfluous and is breaking cephfs tests.
The `std::endl` did not exist earlier.
Is there any specific requirement to have an EOL here or could we change it to `std::flush` ?

Reply:
sorry! whatever it takes to fix the cephfs test. reverting to `cout << s;` should be fine

Reply:
looks like the rados suite caught this too, see https://github.com/ceph/ceph/pull/59025